cpp/src/arrow/compute/kernels/util-internal.cc

```diff
@@ -236,6 +236,43 @@ Status PropagateNulls(FunctionContext* ctx, const ArrayData& input, ArrayData* o
   return Status::OK();
 }
 
+Status PropagateNulls(FunctionContext* ctx, const ArrayData& lhs, const ArrayData& rhs,
```

Comment:
Is this different then AssignNullIntersection?

Reply:
it absolutely is, didn't know why I missed this one. I'll refactor do call AssignNullIntersection or remove it?

Reply:
I think refactoring to call AssignNullIntersection sounds good, since I clearly didn't pick a good name.

Reply:
It's more that I didn't look what was there than name picking. Even worse, I think I even reviewed your function...